test_sanitize.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+"""Tests for git config sanitization to prevent RCE attacks."""
+
+import os
+import re
+import sys
+import tempfile
+import unittest
+
+
+def printf(fmt, *args, file=sys.stdout):
+    if args:
+        fmt = fmt % args
+    file.write(fmt)
+    file.flush()
+
+
+def sanitize_file(filepath):
```

Comment:
don't copy paste the code from `git_dumper.py`. Instead, import git dumper as a module.
And same comment as other PR: please move these in a test directory